src/main/java/com/github/Anon8281/universalScheduler/foliaScheduler/FoliaScheduler.java

```diff
@@ -191,7 +196,42 @@ public void cancelTasks(Plugin plugin) {
         asyncScheduler.cancelTasks(plugin);
     }
 
+    @Override
+    public void cancel(int taskId) {
+        tasks.get(taskId).cancel();
+    }
+
+    @Override
+    public void cancel(MyScheduledTask task) {
+        task.cancel();
+    }
+
     private long getOneIfNotPositive(long x) {
         return x <= 0 ? 1L : x;
     }
+
+    private synchronized int cleanAndGetNextId() {
+        int taskId;
+
+        handleClean();
+
+        do {
+            taskId = new Random().nextInt();
+        } while (tasks.containsKey(taskId));
```

Comment:
If we don’t use random, we will still be forced to use numbers until we find a free task ID.